### PR TITLE
Enable mixer tests in pilot e2e

### PR DESCRIPTION
**Pull request by mandarjog**
1. Enables mixer tests in pilot e2e
2. Some minor modifications to tests

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -147,7 +147,7 @@ jobs:
       - run: make docker.prebuilt
       - run: bin/testEnvRootMinikube.sh wait
       - run: docker images
-      - run: $GOPATH/bin/pilot-integration-test --logtostderr -hub $HUB -tag $TAG -mixer=false -auth=enable -errorlogsdir=/home/circleci/logs -use-initializer
+      - run: $GOPATH/bin/pilot-integration-test --logtostderr -hub $HUB -tag $TAG -mixer=true -auth=enable -errorlogsdir=/home/circleci/logs -use-initializer
```
Comment (kyessenov) [inline .circleci/config.yml]: just remove `-mixer` argument

File: pilot/test/integration/driver.go [hunk 2]
```diff
@@ -277,12 +278,11 @@ func runTests(envs ...infra) {
...
 		}
 
-		// always remove infra even if the tests fail
-		tlog("Tearing down infrastructure", istio.Name)
-		istio.teardown()
 
 		if errs == nil {
 			tlog("Passed all tests!", fmt.Sprintf("tests: %v, count: %d", tests, count))
+			tlog("Tearing down infrastructure", istio.Name)
+			istio.teardown()
```
Comment (kyessenov) [inline pilot/test/integration/driver.go]: always tear down, or we will be leaking resources

File: pilot/test/util/shell.go [hunk 3]
```diff
@@ -39,7 +39,11 @@ func Run(command string) error {
 
 // RunInput command and pass input via stdin
 func RunInput(command, input string) error {
-	log.Infof("Run %q on input:\n%s", command, input)
+	clipped := input
+	if len(clipped) > 20 {
+		clipped = fmt.Sprintf("%s <clipped len=%d>", clipped[0:20], len(input))
```
Comment (kyessenov) [inline pilot/test/util/shell.go]: ugh, XML in console? :)

File: .circleci/config.yml (hunk 1 above)
Comment (rshriram) [reply .circleci/config.yml]: No please keep it. It helps debug things later.. and one does not have to dig through the code to debug.

Comment (kyessenov) [reply .circleci/config.yml]: `pilot-integration-test --help` is your friend

**Pull request merged**